File: API/Features/Reservations/ViewModels/ReservationListVM.cs
using System;
using System.Collections.Generic;

namespace API.Features.Reservations {

    public class ReservationListVM {

        public Guid ReservationId { get; set; }
        public ReservationListBoatVM Boat { get; set; }
        public IEnumerable<ReservationListBerthVM> Berths { get; set; }
        public string FromDate { get; set; }
        public string ToDate { get; set; }
        public int Days { get; set; }
        public bool IsAthenian { get; set; }
        public bool IsFishingBoat { get; set; }
        public bool IsDocked { get; set; }
        public bool IsDryDock { get; set; }
        public bool IsPassingBy { get; set; }
        public bool IsOverdue { get; set; }

    }

}

File: FrontEnd/src/app/features/reservations/classes/view-models/reservation-list-vm.ts
import { ReservationListBerthVM } from './reservation-list-berth-vm'
import { ReservationListBoatVM } from './reservation-list-boat-vm'

export interface ReservationListVM {

    reservationId: string
    boat: ReservationListBoatVM
    fromDate: string
    toDate: string
    berths: ReservationListBerthVM[]
    joinedBerths: string
    isAthenian: boolean
    isFishingBoat: boolean
    isDocked: boolean
    isDryDock: boolean
    isPassingBy: boolean
    isOverdue: boolean

}
